backend/src/feedback/generateLink.test.ts

```diff
@@ -0,0 +1,69 @@
+import { prisma } from '../server/prisma';
+import { resolverHelper } from '../graphql/resolverHelper';
+import {
+  generateLink as generateLinkResolver,
+  GENERATE_LINK_DAY_LIMIT,
+} from './generateLink';
+import mail from '@sendgrid/mail';
+import { EMAIL_SENDER, FEEDBACK_TEMPLATE_ID, REPLY_TO } from '../utils/constants';
+
+const generateLink = resolverHelper(generateLinkResolver);
+jest.mock('@sendgrid/mail');
+jest.mock('nanoid', () => {
+  let x = 0;
+  return {
+    nanoid: () => 'SOMELONGRANDOMTEXT' + x++,
+  };
+});
+jest.mock('./utils/feedbackUniqueId');
+
+describe('generateLink', () => {
+  afterEach(async () => {
+    await prisma.feedbackBase.deleteMany();
+    await prisma.user.deleteMany();
+  });
+
+  it('creates a user and a link and sends it to the mail address', async () => {
+    const email = 'test@test.test';
+    const result = await generateLink({ data: { email } });
+    const userRows = await prisma.user.findMany({});
+    expect(userRows.length).toEqual(1);
+    const userRow = userRows.pop();
+    expect(userRow?.email).toEqual(email);
+
+    const feedbackBaseRows = await prisma.feedbackBase.findMany({});
+    expect(feedbackBaseRows.length).toEqual(1);
+
+    const feedbackBaseRow = feedbackBaseRows.pop();
+    expect(feedbackBaseRow?.feedbackUid).toEqual('xxx-yyy-zzzz0');
+    expect(feedbackBaseRow?.reportUid).toEqual('SOMELONGRANDOMTEXT0');
+    expect(feedbackBaseRow?.userId).toEqual(userRow?.id);
+
+    expect(mail.send).toBeCalledTimes(1);
+    expect(mail.send).toBeCalledWith({
+      dynamicTemplateData: {
+        feedback: process.env.CLIENT_ADDRESS + '/feedback/xxx-yyy-zzzz0',
+        report: process.env.CLIENT_ADDRESS + '/report/SOMELONGRANDOMTEXT0',
+      },
+      from: { email: EMAIL_SENDER, name: 'Miten.io' },
+      reply_to: REPLY_TO,
+      templateId: FEEDBACK_TEMPLATE_ID,
+      to: email,
+    });
+
+    expect(result?.success).toEqual(true);
+  });
+
+  it('does not generate more than limit for a single mail address', async () => {
+    const email = 'test@test.test';
```

Comment:
Nitpick again :D 

example.com is the only official test domain. As a best practice we can always use that in tests, so that one day mistakenly we don't send test emails to an actual inbox :)  [100% sure that it's not related to what is happening here, I just feel good about myself when I annoy people with example.com, sorry :D]

Reply:
Right, I'll update all of the emails now.